docs/src/test/scala/sample/scaladsl/ConsumerExample.scala

```diff
@@ -191,6 +185,57 @@ object AtLeastOnceWithBatchCommitExample extends ConsumerExample {
   def business(key: String, value: Array[Byte]): Future[Done] = ???
 }
 
+// Consume messages at-least-once, and commit in batches
+object PassThroughFlowCommitExample extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+
+    // #passThroughFlow
+    var writeFlow = Flow[ConsumerRecord[String, Array[Byte]]].map(_ => ???)
+
+    val control =
+      Consumer
+        .committableSource(consumerSettings, Subscriptions.topics("topic1"))
+        .map(msg => (msg.record, msg.committableOffset))
```

Comment:
Alternatively, the user could pass a selector function which extracts the part used for the inner flow from the generic in-data type.

Reply:
Sorry, here I don't understand what do you mean ... In any case feel free to just modify it as you think it more correct, or move in another part of the doc.

Reply:
You've seen the function in the HBase connector, converting the incoming data to mutations. Here the user could provide a function to select the data to be passed to the flow from the incoming message. 
The Graph DSL would `broadcast`, apply the selector function, and pass the result to the flow.
On the outgoing side, you would keep the `zip`.